oauth.go

```diff
@@ -50,7 +50,7 @@ func (app App) GetAccessToken(shopName string, code string) (string, error) {
 	req, err := client.NewRequest("POST", "admin/oauth/access_token", data, nil)
 
 	token := new(Token)
-	err = client.Do(req, token)
+	err = client.Do(req, token, nil)
```

Comment:
Should check the error and return a nil token if the error is not nil. Unless you state otherwise the common practice is to return nil or empty values whenever there is an error returned.

Reply:
In this instance I only added the `nil` argument, everything else was already like that.
I'm happy to put an error check here, there should also be one after `NewRequest()` above

Reply:
This actually seems to be common around the codebase.
e.g.
https://github.com/bold-commerce/go-shopify/blob/master/asset.go#L60-L61
https://github.com/bold-commerce/go-shopify/blob/master/product.go#L123-L124

Reply:
All good then. I mean it's wrong in general go but if the entire code base is doing it's ok I guess. More of a style thing to avoid bugs with forgetting to check errors and slight perf of not needing return values when there is an error.

Reply:
Makes sense. I'll make sure I do it in code I touch but not retrospectively add it.